tiledb/sm/array_schema/dimension.cc

```diff
@@ -1379,62 +1379,13 @@ Status Dimension::set_tile_extent(const ByteVecValue& tile_extent) {
 }
 
 Status Dimension::set_null_tile_extent_to_range() {
-  switch (type_) {
-    case Datatype::INT8:
-      return set_null_tile_extent_to_range<int8_t>();
-    case Datatype::UINT8:
-      return set_null_tile_extent_to_range<uint8_t>();
-    case Datatype::INT16:
-      return set_null_tile_extent_to_range<int16_t>();
-    case Datatype::UINT16:
-      return set_null_tile_extent_to_range<uint16_t>();
-    case Datatype::INT32:
-      return set_null_tile_extent_to_range<int32_t>();
-    case Datatype::UINT32:
-      return set_null_tile_extent_to_range<uint32_t>();
-    case Datatype::INT64:
-      return set_null_tile_extent_to_range<int64_t>();
-    case Datatype::UINT64:
-      return set_null_tile_extent_to_range<uint64_t>();
-    case Datatype::FLOAT32:
-      return set_null_tile_extent_to_range<float>();
-    case Datatype::FLOAT64:
-      return set_null_tile_extent_to_range<double>();
-    case Datatype::DATETIME_YEAR:
-    case Datatype::DATETIME_MONTH:
-    case Datatype::DATETIME_WEEK:
-    case Datatype::DATETIME_DAY:
-    case Datatype::DATETIME_HR:
-    case Datatype::DATETIME_MIN:
-    case Datatype::DATETIME_SEC:
-    case Datatype::DATETIME_MS:
-    case Datatype::DATETIME_US:
-    case Datatype::DATETIME_NS:
-    case Datatype::DATETIME_PS:
-    case Datatype::DATETIME_FS:
-    case Datatype::DATETIME_AS:
-    case Datatype::TIME_HR:
-    case Datatype::TIME_MIN:
-    case Datatype::TIME_SEC:
-    case Datatype::TIME_MS:
-    case Datatype::TIME_US:
-    case Datatype::TIME_NS:
-    case Datatype::TIME_PS:
-    case Datatype::TIME_FS:
-    case Datatype::TIME_AS:
-      return set_null_tile_extent_to_range<int64_t>();
-    case Datatype::STRING_ASCII:
+  auto g = [&](auto T) {
```

Comment:
We shouldn't get rid of the default case here... See my comment below about using concepts...